pkg/apis/crd/v1alpha1/types.go

```diff
@@ -709,3 +709,127 @@ type ExternalNodeList struct {
+}
+
+type SupportBundleCollectionConditionType string
+
+const (
+	// CollectionStarted is added in a SupportBundleCollection when Antrea Controller has started to handle the request.
+	CollectionStarted SupportBundleCollectionConditionType = "Started"
+	// CollectionCompleted is added in a SupportBundleCollection when Antrea has finished processing the collection.
+	CollectionCompleted SupportBundleCollectionConditionType = "Completed"
+	// CollectionFailure is added in a SupportBundleCollection when one of its required Nodes/ExternalNodes fails
+	// to process the request.
+	CollectionFailure SupportBundleCollectionConditionType = "CollectionFailure"
+	// BundleCollected is added in a SupportBundleCollection when at least one of its required Nodes/ExternalNodes
+	// successfully uploaded files to the file server.
+	BundleCollected SupportBundleCollectionConditionType = "BundleCollected"
+)
+
+// SupportBundleCollectionCondition describes the state of a SupportBundleCollection at a certain point.
+type SupportBundleCollectionCondition struct {
+	// Type of StatefulSet condition.
+	Type SupportBundleCollectionConditionType `json:"type"`
+	// Status of the condition, one of True, False, Unknown.
+	Status metav1.ConditionStatus `json:"status"`
+	// Last time the condition transitioned from one status to another.
+	// +optional
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty"`
+	// The reason for the condition's last transition.
+	// +optional
+	Reason string `json:"reason,omitempty"`
+	// A human-readable message indicating details about the transition.
+	// +optional
+	Message string `json:"message,omitempty"`
+}
+
+type BundleNodes struct {
+	// List the names of certain Nodes which are expected to collect and upload
+	// bundle files.
+	// +optional
+	NodeNames []string `json:"nodeNames,omitempty"`
+	// Select certain Nodes which match the label selector.
+	// +optional
+	NodeSelector *metav1.LabelSelector `json:"nodeSelector,omitempty"`
+}
+
+type BundleExternalNodes struct {
+	Namespace string
+	// List the names of certain ExternalNodes which are expected to collect and upload
+	// bundle files.
+	// +optional
+	NodeNames []string `json:"nodeNames,omitempty"`
+	// Select certain ExternalNodes which match the label selector.
+	// +optional
+	NodeSelector *metav1.LabelSelector `json:"nodeSelector,omitempty"`
+}
+
+// BundleFileServer specifies the bundle file server information.
+type BundleFileServer struct {
+	// The URL of the bundle file server. It is set with format: scheme://host[:port][/path],
+	// e.g, https://api.example.com:8443/v1/supportbundles/. If scheme is not set, https is used by default.
+	URL string `json:"url"`
+}
+
+// BundleServerAuthType defines the authentication type to access the BundleFileServer.
+type BundleServerAuthType string
+
+const (
+	APIKey      BundleServerAuthType = "apiKey"
+	BearerToken BundleServerAuthType = "bearerToken"
```

Comment:
The convention is to use UpperCamelCase for enum values, e.g. the image policy `IfNotPresent`, the direction `Ingress`, `Egress`

Reply:
updated.